k8s/upgrade-job/src/upgrade/data_plane.rs

```diff
@@ -211,9 +271,29 @@ async fn is_data_plane_pod_running(
                 .build(),
             )?
             .as_str();
+
+        // skip if the the node name is not the same as the one on which pod needs to be restarted.
         if node != node_name {
             continue;
         } else {
+            // uid of io engine pod on the same node.
+            let new_io_engine_pod_uid = pod.metadata.uid.as_ref().ok_or(
+                EmptyPodUid {
+                    name: pod.name_any(),
+                    namespace: namespace.clone(),
+                }
+                .build(),
+            )?;
+
+            // validating that the pod uid on the node is same or not
+            // if same, that means that pod has not been started yet
+            // and new data plane pod is not up and running.
+            if old_io_engine_pod_uid.eq(new_io_engine_pod_uid) {
+                let pod_name = pod.name_any();
+                tracing::warn!(node.name = %node, pod.name = %pod_name, "Pod has not been deleted yet.");
```

Comment:
```suggestion
                tracing::warn!(node.name = %node, pod.name = %pod_name, "Pod has not been deleted yet");
```
traces should not end with `.`